### PR TITLE
Using PNGFiles.jl via ImageIO.jl (lazy loader) for .png image files

### PR DESCRIPTION
**Pull request by ianshmean**
https://github.com/JuliaIO/PNG.jl isn't yet registered, but is functional with this branch.

### CODE REVIEW

**Comment by ianshmean** [comment]
As @davidanthoff pointed out on slack, we need to make  sure the MIME for png isn't messed  up by this. Just making a note as a reminder

**Comment by timholy** [comment]
Need to wait for https://github.com/JuliaRegistries/General/pull/9508 (3 days).

**Comment by johnnychen94** [comment]
Does this mean that to load a png file we'll need to install `PNG` instead of `ImageMagick`? If so then a lot of packages' test codes will be broken without adding `PNG` to the `Project.toml`'s `extra` section.

**Comment by ianshmean** [comment]
> Need to wait for JuliaRegistries/General#9508 (3 days).

Indeed, and as @johnnychen94's point illustrates, even once that's registered this needs careful thinking before merging

**Comment by ianshmean** [comment]
It makes me think that if we were to move to this approach, it might be worth waiting until  the big three, PNG.jl, JPEG.jl and TIFF.jl, were ready to make the switchover easier?

**Comment by timholy** [comment]
IIRC packages get precedence by argument order. So we could leave QuartzImageIO and ImageMagick as valid loaders but put PNG as the first choice.

**Comment by johnnychen94** [comment]
If the final goal is to deprecate ImageMagick(is it?), then I think the following procedure makes sense:

1. add a deprecation warning in `ImageMagick` and `QuartzImageIO` for specific image format to encourage users to try out `PNG.jl`
2. leave `ImageMagick` and `QuartzImageIO` as fallbacks in `FileIO` for a period
3. unregister `ImageMagick` and `QuartzImageIO` from `FileIO`

**Comment by timholy** [comment]
I think we're likely to have ImageMagick around for some time for less commonly-used formats. Just restoring those lines to the `add_format` seems like it should be enough even without changing any of the other packages.

File: src/registry_setup.jl [hunk 1]
```diff
@@ -81,7 +81,7 @@ end
 `add_format(fmt, magic, extention)` registers a new `DataFormat`.
 For example:
 
-    add_format(format"PNG", (UInt8[0x4d,0x4d,0x00,0x2b], UInt8[0x49,0x49,0x2a,0x00]), [".tiff", ".tif"])
+    add_format(format"TIFF", (UInt8[0x4d,0x4d,0x00,0x2b], UInt8[0x49,0x49,0x2a,0x00]), [".tiff", ".tif"])
```
Comment (ianshmean) [inline src/registry_setup.jl]: Just wanted to highlight that I thought this was a typo, so corrected it

**Comment by davidanthoff** [comment]
Could you try whether this would break the existing MIME based PNG story? Here is a simple test:
```
using VegaLite, VegaDatasets

dataset("cars") |> @vlplot(:point, x=:Acceleration) |> save("foo.png")
```

**Comment by ianshmean** [comment]
@davidanthoff does this look right?
foo

**Comment by davidanthoff** [comment]
Yes, if it produces a proper graph, all is good :)

**Comment by timholy** [comment]
Just strip the WIP out of the title if you're ready for this to be merged.

**Comment by ianshmean** [comment]
Will do. Need to wait for ImageIO to be released and a plan to be settled on for the test failures in https://github.com/JuliaIO/PNGFiles.jl/pull/5

**Comment by ianshmean** [comment]
@timholy Ready to merge, given https://github.com/JuliaIO/PNGFiles.jl/pull/7 has been merged

**Comment by timholy** [comment]
Oh boy oh boy oh boy!

**Pull request merged**